### ISSUE TITLE
Enhancement genhtml : Increase contrast of text (a11y)

### ISSUE THREAD

**Issue by IzawGithub** [linux-test-project/lcov#214]
Report produced by genhtml can have a low contrast. This is mainly noticeable when using the `--dark-mode` option, where `tableHead` text is black on grayish blue, which gives a poor WCAG contrast of 2,46 . This makes text hard to read, especially if one has vision problem.

This issue does not only affect the dark-mode table header, as another notable contrast issue is the light-mode visited link, with a contrast ratio of 1,75 . There are probably more place that I missed where the contrast is subpar, or could be improved for better readability.

Here is a quick demo of a possible improvement.

Light-mode
![LCovLight](https://github.com/linux-test-project/lcov/assets/125356732/7e733b06-6434-44e7-8f3e-1e8cf71dd02d)

Dark-mode
![LCovDark](https://github.com/linux-test-project/lcov/assets/125356732/b99eb1ac-14a4-4274-b1d5-01d051492443)

I do not know Perl, nor do I know how to build the project, so I cannot create a pull request to implement this change.

**Comment by IzawGithub** [comment]
Hello, thanks for the quick response! I'll document what are some of the css class that need better contrast, and I'll try to make a pull request with some more readable color.

Keep in mind that I opened an issue instead of a pull request, because my UI design skill are also quite lackluster. The best I can do is a color wheel and changing until the contrast is better 😅

**Comment by henry2cox** [comment]
If we are making changes, it would be good idea to pass suggestions by someone with red/green and blue/yellow color blindness.
Of the two:  red/green is considerably more common.

If you or somebody else was interested enough to suggest a full palette for those individuals, it would be pretty easy to add an option to the tool so that the user could select the one that works best for them (assuming they didn't want to override the CSS).

**Comment by IzawGithub** [comment]
Hello. Regarding color blindness, a possible solution would be the introduction of an `--high-contrast` mode to genhtml, that would turn all color to only shades of gray.
This has the advantage of fixing every existing sort of color blindness, while also improving readability for user preferring usability over style.

I will work on contrast for the existing style during this weekend.

**Comment by henry2cox** [comment]
greyscale is a good idea.  Had not thought of that.
I think we would have to limit the number of levels that we try to use, though (not try for all 12 differential categories).
I suspect that you would need pretty good visual acuity to distinguish more than 6 or 8 levels (...somebody has definitely studied that).

**Comment by hramrach** [comment]
There is also the simulator https://www.color-blindness.com/coblis-color-blindness-simulator/

Not sure how realistic it is. Nonetheless, the grayscale view has most problems, and is indeed unreadable because of low contrast.

**Comment by henry2cox** [comment]
over a year with no followup - so I am going to close this now.
Please feel free to reopen or file a new issue or PR when/if this becomes important again.
